test-specifications/token-service/tokenUpdateTransaction.md

```diff
@@ -0,0 +1,810 @@
+| 4       | Updates a mutable token with a valid ED25519 private key as its metadata key                                                | tokenId=<CREATED_MUTABLE_TOKEN_ID>, metadataKey=<VALID_ED25519_PRIVATE_KEY>, commonTransactionParams.signers=[<CREATED_MUTABLE_TOKEN_ADMIN_KEY>]          | The token update succeeds and the token has the new ED25519 private key as its pause key.        | N                 |
+| 5       | Updates a mutable token with a valid ECDSAsecp256k1 private key as its metadata key                                         | tokenId=<CREATED_MUTABLE_TOKEN_ID>, metadataKey=<VALID_ECDSA_SECP256K1_PRIVATE_KEY>, commonTransactionParams.signers=[<CREATED_MUTABLE_TOKEN_ADMIN_KEY>]  | The token update succeeds and the token has the new ECDSAsecp256k1 private key as its pause key. | N                 |
+| 6       | Updates a mutable token with a valid KeyList of ED25519 and ECDSAsecp256k1 private and public keys as its metadata key      | tokenId=<CREATED_MUTABLE_TOKEN_ID>, metadataKey=<VALID_KEYLIST>, commonTransactionParams.signers=[<CREATED_MUTABLE_TOKEN_ADMIN_KEY>]                      | The token update succeeds and the token has the new KeyList as its pause key.                    | N                 |
+| 7       | Updates a mutable token with a valid KeyList of nested Keylists (three levels) as its metadata key                          | tokenId=<CREATED_MUTABLE_TOKEN_ID>, metadataKey=<VALID_NESTED_KEYLIST>, commonTransactionParams.signers=[<CREATED_MUTABLE_TOKEN_ADMIN_KEY>]               | The token update succeeds and the token has the new nested KeyList as its pause key.             | N                 |
+| 8       | Updates a mutable token with a valid ThresholdKey of ED25519 and ECDSAsecp256k1 private and public keys as its metadata key | tokenId=<CREATED_MUTABLE_TOKEN_ID>, metadataKey=<VALID_THRESHOLD_KEY>, commonTransactionParams.signers=[<CREATED_MUTABLE_TOKEN_ADMIN_KEY>]                | The token update succeeds and the token has the new ThresholdKey as its pause key.               | N                 |
+| 9       | Updates a mutable token with a valid key as its metadata key but doesn't sign with it                                       | tokenId=<VALID_TOKEN_ID_WITHOUT_METADATA_KEY>, metadataKey=<VALID_KEY>, commonTransactionParams.signers=[<ADMIN_KEY_OF_VALID_TOKEN_WITHOUT_METADATA_KEY>] | The token update fails with an TOKEN_HAS_NO_METADATA_KEY response code from the network.         | N                 |
+| 10      | Updates a mutable token with an invalid key as its metadata key                                                             | tokenId=<CREATED_MUTABLE_TOKEN_ID>, metadataKey=<INVALID_KEY>, commonTransactionParams.signers=[<CREATED_MUTABLE_TOKEN_ADMIN_KEY>]                        | The token update fails with an SDK internal error.                                               | N                 |
```

Comment:
We may want to add a test case to update the token with an empty `KeyList` as the metadata key.  That's how a "role" key can be removed in a token update.

We might want to include that in each of the "role" keys for update, to cover that "clear a key" behavior.

Reply:
Ah good to know, I wasn't aware that's how it worked. I added a test for each "role" key to test this behavior.